File: java/src/test/java/problems1_100/RemoveElementTest.java
package problems1_100;

import java.util.Arrays;
import org.junit.Test;
import sample.SampledTest;

import static com.google.common.truth.Truth.assertThat;
import static com.google.common.truth.Truth.assertWithMessage;

public class RemoveElementTest extends SampledTest {
    @Test
    public void test() {
        for (RemoveElement solution : RemoveElement.values()) {
            for (RemoveElementSample sample : getSamples(RemoveElementSample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(
                        solution.removeElement(
                            sample.input.nums,
                            sample.input.val
                        )
                    ).isEqualTo(sample.output);
                assertThat(
                    Arrays
                        .stream(sample.input.nums)
                        .boxed()
                        .toList()
                        .subList(0, sample.output)
                ).doesNotContain(sample.input.val);
            }
        }
    }
}

File: kotlin/src/test/kotlin/problems1_100/RemoveElementTest.kt
package problems1_100

import com.google.common.truth.Truth.assertThat
import com.google.common.truth.Truth.assertWithMessage
import sample.SampledTest
import sample.component1
import sample.component2
import sample.component3
import sample.getSamples
import kotlin.test.Test

class RemoveElementTest : SampledTest() {
    @Test
    fun test() =
        RemoveElement.entries.forEach { solution ->
            getSamples<Array<RemoveElementSample>>().forEach { (input, output, message) ->
                assertWithMessage(message)
                    .that(solution.removeElement(input.nums, input.`val`))
                    .isEqualTo(output)
                assertThat(input.nums.toList().subList(0, output))
                    .doesNotContain(input.`val`)
            }
        }
}
